src/main/java/frc/robot/subsystems/vision/SImVisionIO.java

```diff
@@ -35,29 +38,35 @@ private SImVisionIO(int camIndex) {
         }
     }
 
-
     @Override
     public void setPipeLine(int pipeLineIndex) {
         simCamera.getCamera().setPipelineIndex(pipeLineIndex);
     }
 
     @Override
     public Result getLatestResult() {
-        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(),simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
+        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(), simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
     }
 
     @Override
-    public void updateInputs(VisionInputs inputs) {
+    public void updateInputs(VisionInputsAutoLogged inputs) {
 
         for (int i = 1; i < VisionConstants.TARGET_POSITION_SIM.length; i++) {
             coprocessorSim.addVisionTargets(VisionConstants.SIM_VISION_TARGETS[i]);
             coprocessorSim.update(SwerveDrive.getInstance().getBotPose());
+            estimatedPose = estimator.update(simCamera.getCamera().getLatestResult()).get().estimatedPose;
         }
-        estimator.update();
-
 
+        inputs.hasTargets = simCamera.getCamera().getLatestResult().hasTargets();
+        inputs.yaw = simCamera.getCamera().getLatestResult().getBestTarget().getYaw();
+        inputs.pitch = simCamera.getCamera().getLatestResult().getBestTarget().getPitch();
+        inputs.area = simCamera.getCamera().getLatestResult().getBestTarget().getArea();
+        inputs.targetSkew = simCamera.getCamera().getLatestResult().getBestTarget().getSkew();
+        inputs.targetID = simCamera.getCamera().getLatestResult().getBestTarget().getFiducialId();
```

Comment:
Here too `simCamera.getCamera().getLatestResult().getBestTarget()` needs to be extracted in a variable. In IntelliJ IDEA, you can select this text and use <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>V</kbd> to replace all usages with the new variable.